saleor/graphql/giftcard/mutations.py

```diff
@@ -0,0 +1,153 @@
+        if balance:
+            cleaned_input["current_balance"] = balance
+            cleaned_input["initial_balance"] = balance
+        buyer_email = data.get("buyer_email", None)
+        if buyer_email:
+            try:
+                cleaned_input["buyer"] = User.objects.get(email=buyer_email)
+            except ObjectDoesNotExist:
+                raise ValidationError(
+                    {"email": "Customer with this email doesn't exist."}
+                )
+        return cleaned_input
+
+    @classmethod
+    def save(cls, info, instance, cleaned_input):
+        instance.creator = info.context.user
+        super().save(info, instance, cleaned_input)
+
+
+class GiftCardUpdate(GiftCardCreate):
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a gift card to update.")
+        input = GiftCardInput(
+            required=True, description="Fields required to update a gift card."
+        )
+
+    class Meta:
+        description = "Update a gift card."
+        model = models.GiftCard
+        permissions = ("giftcard.manage_gift_card",)
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        code = data.get("code", None)
+        if code:
+            raise ValidationError({"code": "Cannot update a gift card code."})
+        return super().clean_input(info, instance, data)
+
+
+class GiftCardDeactivate(BaseMutation):
+    gift_card = graphene.Field(GiftCard, description="A gift card to deactivate.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a gift card to deactivate.")
+
+    class Meta:
+        description = "Deactivate a gift card."
+        permissions = ("giftcard.manage_gift_card",)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        gift_card_id = data.get("id")
+        gift_card = cls.get_node_or_error(
+            info, gift_card_id, field="gift_card_id", only_type=GiftCard
+        )
+        if gift_card.is_active:
+            gift_card.is_active = False
+            gift_card.save(update_fields=["is_active"])
+        return GiftCardDeactivate(gift_card=gift_card)
+
+
+class GiftCardActivate(BaseMutation):
+    gift_card = graphene.Field(GiftCard, description="A gift card to activate.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a gift card to activate.")
+
+    class Meta:
+        description = "Activate a gift card."
+        permissions = ("giftcard.manage_gift_card",)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        gift_card_id = data.get("id")
+        gift_card = cls.get_node_or_error(
+            info, gift_card_id, field="gift_card_id", only_type=GiftCard
+        )
+        if not gift_card.is_active:
+            gift_card.is_active = True
+            gift_card.save(update_fields=["is_active"])
+        return GiftCardDeactivate(gift_card=gift_card)
+
+
+class GiftCardVerify(BaseMutation):
```

Comment:
I think this whole mutation is not needed. Gift cards would be verified automatically when adding them to checkout.